tox.ini

```
@@ -95,13 +95,12 @@ deps =
     bandit[toml]
     -r{toxinidir}/requirements.txt
 commands =
-    bandit -c {toxinidir}/pyproject.toml -r {[vars]src_path} {[vars]tst_path}
+    bandit -c {toxinidir}/pyproject.toml -r {[vars]app_path} {[vars]tst_path}
 
 [testenv:integration]
 description = Run integration tests
 deps =
-    # Last compatible version with Juju 2.9
-    juju==3.1.2.0
+    juju==3.1.0.1
```

Out of curiosity, I'm wondering if there is a reason to pin it to this specific version or we could do
```suggestion
    juju>=3.1
```

---

I did this because we decided to pin as specifically as possible (see internal spec ISD 115 and also a reference tox which refers to https://github.com/canonical/mysql-router-k8s-operator/blob/d89b0f820cfc574b04e0b86b5412db9da0d7ff2e/pyproject.toml#L60 which also pins quite strictly). 
However, the spec mainly refers to the artefacts produced and not the test environment, so your suggestion may be fine.